lib/services/unified_resource.go

```diff
@@ -184,23 +188,26 @@ func (c *UnifiedResourceCache) deleteSortKey(sortKey resourceSortKey) error {
 // delete removes the item by key, returns NotFound error
 // if item does not exist
 func (c *UnifiedResourceCache) delete(ctx context.Context, res types.Resource) error {
-	key := resourceKey(res)
+	c.rw.Lock()
+	defer c.rw.Unlock()
+	if c.stale {
+		return nil
+	}
```

Comment:
This is a new bypass, is it an intended change?

Reply:
It's actually preservation of old behavior that I wasn't sure was intended or not so I decided to leave in.  Looking closer, turns out this function isn't actually necessary anymore, so I've just sidestepped the question by removing it.